src/datasets/datasets.service.ts

```diff
@@ -41,22 +43,33 @@ import { DatasetClass, DatasetDocument } from "./schemas/dataset.schema";
 
 @Injectable({ scope: Scope.REQUEST })
 export class DatasetsService {
+  private ESClient: ElasticSearchService | null;
   constructor(
     private configService: ConfigService,
     @InjectModel(DatasetClass.name)
     private datasetModel: Model<DatasetDocument>,
     private initialDatasetsService: InitialDatasetsService,
     private logbooksService: LogbooksService,
+    @Inject(ElasticSearchService)
+    private elasticSearchService: ElasticSearchService,
     @Inject(REQUEST) private request: Request,
-  ) {}
+  ) {
+    if (this.elasticSearchService.connected) {
+      this.ESClient = this.elasticSearchService;
+    }
+  }
 
   async create(createDatasetDto: CreateDatasetDto): Promise<DatasetDocument> {
     const username = (this.request.user as JWTUser).username;
     const createdDataset = new this.datasetModel(
       // insert created and updated fields
       addCreatedByFields(createDatasetDto, username),
     );
-
+    if (this.ESClient) {
+      this.ESClient.insertDocument(
+        createdDataset.toObject() as DatasetDocument,
+      );
+    }
```

Comment:
How do we change which dataset fields we are inserting in elastic search?

Reply:
```
  async updateDocument(data: DatasetDocument) {
    delete data._id;
    try {
      await this.esService.index({
        index: this.defaultIndex,
        id: data.pid,
        document: data,
      });
```
I think we need to specify here, let the `document: ` to only update given field.
Should be feasible to update only specific field

Reply:
WE should specify the document fields that will be inserted in ES in scicat BE configuration.
Let's open a new issue regarding this topic